lib/charms/mongodb_libs/v0/mongodb_provider.py

```diff
@@ -133,6 +150,143 @@ def oversee_users(self, departed_relation_id: Optional[int], event):
                 logger.info("Drop database: %s", database)
                 mongo.drop_database(database)
 
+    ################################################################################
+    # LEGACY RELATIONS
+    ################################################################################
+    def _on_legacy_relation_created(self, event):
+        """Legacy relations for MongoDB operate without a password and so we update the server accordingly and
+        set a flag.
+        """
+        logger.warning("DEPRECATION WARNING - `mongodb` interface is a legacy interface.")
+
+        # TODO, future PR check if there are any new relations that are related to this charm and will be effected
+        # by loss of password. If so go into blocked state.
+
+        # TODO, future PR check if already running without auth
+        try:
+            self._stop_mongod_service()
+            self._update_mongod_service(auth=False)
+            self._start_mongod_service()
+        except systemd.SystemdError:
+            self.charm.unit.status = BlockedStatus("couldn't restart MongoDB")
+            return
+
+    def _on_legacy_relation_joined(self, event):
+        """
+        NOTE: this is retro-fitted from the legacy mongodb charm: https://launchpad.net/charm-mongodb
+        """
+        logger.warning("DEPRECATION WARNING - `mongodb` interface is a legacy interface.")
+        if not self.charm.unit.is_leader:
```

Comment:
i believe that `is_leader` is a method and not a property ([see here](https://github.com/canonical/operator/blob/4ffc12569dc576d2a9f9f44269b77616b15dc430/ops/model.py#L454))